src/Saunter/AttributeProvider/AttributeDocumentProvider.cs

```diff
@@ -0,0 +1,404 @@
+                {
+                    messages.Add(message);
+                }
+            }
+
+            return operation;
+        }
+
+        private static OperationAttribute? GetOperationAttribute(MemberInfo typeOrMethod, OperationType operationType)
+        {
+            return operationType switch
+            {
+                OperationType.Publish => typeOrMethod.GetCustomAttribute<PublishOperationAttribute>(),
+                OperationType.Subscribe => typeOrMethod.GetCustomAttribute<SubscribeOperationAttribute>(),
+                _ => null,
+            };
+        }
+
+        private List<AsyncApiMessage> GenerateMessageFromAttributes(IEnumerable<MessageAttribute> messageAttributes)
+        {
+            List<AsyncApiMessage> messages = new();
+
+            if (messageAttributes.Count() == 1)
+            {
+                var message = GenerateMessageFromAttribute(messageAttributes.First());
+
+                if (message is not null)
+                {
+                    messages.Add(message);
+                }
+
+                return messages;
+            }
+
+            foreach (var attribute in messageAttributes)
+            {
+                var message = GenerateMessageFromAttribute(attribute);
+
+                if (message != null)
+                {
+                    messages.Add(message);
+                }
+            }
+
+            return messages;
+        }
+
+        private List<AsyncApiMessage> GenerateMessageFromType(TypeInfo payloadType)
+        {
+            var message = new AsyncApiMessage
+            {
+                Payload = _schemaGenerator.Generate(payloadType),
+            };
+
+            message.MessageId = message.Payload?.Title;
+            message.Name = message.Payload?.Title;
+
+            return new() { message };
+        }
+
+        private AsyncApiMessage? GenerateMessageFromAttribute(MessageAttribute messageAttribute)
+        {
+            if (messageAttribute?.PayloadType == null)
+            {
+                return null;
+            }
+
+            var tags = messageAttribute.Tags?
+                .Select(x => new AsyncApiTag { Name = x })
+                .ToList() ?? new List<AsyncApiTag>();
+
+            var asyncApiSchema = _schemaGenerator.Generate(messageAttribute.PayloadType);
+
+            var message = new AsyncApiMessage
+            {
+                MessageId = messageAttribute.MessageId ?? asyncApiSchema?.Title ?? messageAttribute.PayloadType.Name,
+                Title = messageAttribute.Title,
+                Summary = messageAttribute.Summary,
+                Description = messageAttribute.Description,
+                Tags = tags,
+                Payload = asyncApiSchema,
+                Headers = _schemaGenerator.Generate(messageAttribute.HeadersType),
+                Name = messageAttribute.Name ?? asyncApiSchema?.Title ?? messageAttribute.PayloadType.Name,
+                Bindings = new()
```

Comment:
Hi, look. Previously, everything worked on links via 'document.Components', now I've done everything without links. In general, all your questions relate to this)

I can restore the work through links and components, we can leave it as it is, which way will we go?

Reply:
What i mean is; 
You have code that adds things as references (new AsyncApiReference()).
When doing this you have to add whatever that reference is supposed to be, to components, otherwise the document will not be valid. 

In schemagenerator for instance
![image](https://github.com/user-attachments/assets/269d81f2-d544-4a10-96fd-1f74f3438db7)
Whatever that reference is pointing at, needs to exist in the components - but we dont seem to add it anywhere.

Reply:
Oh, I'll fix it now.

There was another problem, I quickly translated everything into links. The LEGO library refused to parse it....

Tell me, is this my mistake or is it a LEGO bug?

Source doc:
![image](https://github.com/user-attachments/assets/e92bc97f-8f8b-4b7f-9c01-2b036eee8474)

Json doc:
![image](https://github.com/user-attachments/assets/e1551539-733e-4c5e-8836-ad61e3915e36)

Reply:
looks like your mistake, but what did you put as the referenceId? 
Basically it should be 
``` csharp
new AsyncApiReference
{
    Type = AsyncAPI.Models.ReferenceType.Message,
    Id = "lightMeasuredEvent"
}
```

Reply:
It was on the channel that the link to the components did not work, I do not understand why, it works in other places.
In the example from asyncapi studio, they don't use channel allocation in components, and we won't?)

The schemas, parameters, and messages are put into components.

> For the convenience of the review, I tried to make everything in 1 commit)

Reply:
alright perfect. think it looks fine.